packages/backend/src/workers/network.js

```
@@ -1,9 +1,310 @@
+
+  if (nodesResponse instanceof RpcError) {
+    return [];
+  }
+
+  const totalPages = nodesResponse.totalPages;
+
+  nodesResponse.nodes.forEach((node) => {
+    nodeList.push(node);
+  });
+
+  page++;
+
+  while (page <= totalPages) {
+    const response = await pocketInstance
+      .rpc(pocketRpcProvider)
+      .query.getNodes(status, undefined, BigInt(0), undefined, page, perPage);
+
+    // Increment page variable
+    page++;
+
+    if (response instanceof RpcError) {
+      page = totalPages;
+      return;
+    }
+
+    response.nodes.forEach((node) => {
+      nodeList.push(node);
+    });
+  }
+
+  return nodesResponse.nodes;
+}
+
+/**
+ * Get Applications data.
+ *
+ * @param {number} status Status of the apps to retrieve.
+ *
+ * @returns {Promise<Application[]>} The applications data.
+ * @async
+ */
+async function getApplications(status) {
+  let page = 1;
+  let applicationList = [];
+
+  const pocketRpcProvider = await getRPCProvider();
+  const perPage = 100;
+  const applicationsResponse = await pocketInstance
+    .rpc(pocketRpcProvider)
+    .query.getApps(status, BigInt(0), undefined, page, perPage);
+
+  // Check for RpcError
+  if (applicationsResponse instanceof RpcError) {
+    return [];
+  }
+
+  // Retrieve the total pages count
+  const totalPages = applicationsResponse.totalPages;
+
+  // Retrieve the app list
+  while (page <= totalPages) {
+    const response = await pocketInstance
+      .rpc(pocketRpcProvider)
+      .query.getApps(status, BigInt(0), undefined, page, perPage);
+
+    // Increment page variable
+    page++;
+
+    // Check for error
+    if (response instanceof RpcError) {
+      page = totalPages;
+      return;
+    }
+    // Add the result to the application list
+    response.applications.forEach((app) => {
+      applicationList.push(app);
+    });
+  }
+
+  return applicationList;
+}
+
+async function getTotalNodesStaked() {
+  const stakedNodes = (await getNodes(StakingStatus.Staked)) ?? [];
```

Not too too sure about this anymore. We really want to throw an error if this fails, or else we might get misleading data on the dashboard.

---

Done in https://github.com/pokt-foundation/pocket-dashboard/pull/27/commits/7e7ea27cfb3ed156d11c25b6a16a83e8a649e6e5